include/revng/Pipeline/Rank.h

```diff
@@ -16,17 +16,28 @@ namespace pipeline {
 /// The rank tree is a tree used by targets to find out how many names
 /// are required to name a target
 class Rank : public DynamicHierarchy<Rank> {
+private:
+  std::string ModelPath;
+
 public:
   Rank(llvm::StringRef Name) : DynamicHierarchy(Name) {}
+  Rank(llvm::StringRef Name, std::string_view ModelPath) :
+    DynamicHierarchy(Name), ModelPath(ModelPath) {}
   Rank(llvm::StringRef Name, Rank &Parent) : DynamicHierarchy(Name, Parent) {}
+  Rank(llvm::StringRef Name, Rank &Parent, std::string_view ModelPath) :
+    DynamicHierarchy(Name, Parent), ModelPath(ModelPath) {}
+
+  std::string_view modelPath() const { return ModelPath; }
 };
 
 // Root rank specialization
-template<ConstexprString String>
+template<ConstexprString String, ConstexprString ModelPathSeparator_>
```

Comment:
What's the use for `ModelPathSeparator_`?

Reply:
This code was written by @Cvelth  , I think it allows for nested `ModelPaths`.

Reply:
Yep, it allows to just specify necessary parts for ranks while automagically merging them with their parent's paths using the separator in between.